src/jetstream/pull_subscription.rs

```diff
@@ -0,0 +1,369 @@
+    ) -> PullSubscription {
+        PullSubscription(Arc::new(Inner {
+            pid,
+            inbox,
+            messages,
+            info: consumer_info,
+            context,
+        }))
+    }
+
+    /// A low level method that should be used only in specific cases.
+    /// Pulls next message available for this `PullSubscription`.
+    /// This operation is blocking and will indefinately wait for new messages.
+    /// Keep in mind that this requires user to request for messages first.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # fn main() -> std::io::Result<()> {
+    /// # let client = nats::connect("demo.nats.io")?;
+    /// # let context = nats::jetstream::new(client);
+    /// #
+    /// # context.add_stream("next")?;
+    /// # context.publish("next", "hello")?;
+    /// # context.add_consumer(
+    /// # "next",
+    /// # nats::jetstream::ConsumerConfig {
+    /// #  durable_name: Some("CONSUMER".to_string()),
+    /// # ..Default::default()
+    /// #  },
+    /// #  )?;
+    /// let consumer = context.pull_subscribe("next", "CONSUMER").unwrap();
+    /// consumer.request_batch(1)?;
+    /// let message = consumer.next();
+    /// println!("Received message: {:?}", message);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn next(&self) -> Option<Message> {
+        self.process_no_messages(self.0.messages.recv().ok())
+    }
+
+    /// A low level method that should be used only in specific cases.
+    /// Pulls next message available for this `PullSubscription`.
+    /// This operation is non blocking, that will yield `None` if there are no messages each time
+    /// it is called.
+    /// Keep in mind that this requires user to request for messages first.
+    ///
+    /// # Example
+    /// ```
+    /// # fn main() -> std::io::Result<()> {
+    /// # let client = nats::connect("demo.nats.io")?;
+    /// # let context = nats::jetstream::new(client);
+    /// #
+    /// # context.add_stream("next")?;
+    /// # context.add_consumer(
+    /// # "next",
+    /// # nats::jetstream::ConsumerConfig {
+    /// #  durable_name: Some("CONSUMER".to_string()),
+    /// # ..Default::default()
+    /// #  },
+    /// #  )?;
+    /// context.publish("next", "hello")?;
+    /// let consumer = context.pull_subscribe("foo", "CONSUMER").unwrap();
+    /// consumer.request_batch(1)?;
+    /// let message = consumer.try_next();
+    /// println!("Received message: {:?}", message);
+    /// assert!(consumer.try_next().is_none());
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn try_next(&self) -> Option<Message> {
+        self.process_no_messages(self.0.messages.try_recv().ok())
+    }
+
+    /// A low level method that should be used only in specific cases.
+    /// Pulls next message available for this `PullSubscription`.
+    /// This operation is contrast to its siblings `next` and `try_next` returns `Message` wrapped
+    /// in `io::Result` as it might return `timeout` error.
```

Comment:
Two kinds of timeouts can occur tho right? there's the actual network timeout or the "wait time" in the request body.

Reply:
fixed docs.